src/Features/CSharp/Portable/UsePatternCombinators/CSharpUsePatternCombinatorsCodeFixProvider.cs

```diff
@@ -0,0 +1,108 @@
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Formatting;
+using Microsoft.CodeAnalysis.Operations;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.UsePatternCombinators
+{
+    using static SyntaxFactory;
+    using static AnalyzedPattern;
+
+    [ExportCodeFixProvider(LanguageNames.CSharp), Shared]
+    internal class CSharpUsePatternCombinatorsCodeFixProvider : SyntaxEditorBasedCodeFixProvider
+    {
+        [ImportingConstructor]
+        [SuppressMessage("RoslynDiagnosticsReliability", "RS0033:Importing constructor should be [Obsolete]", Justification = "Used in test code: https://github.com/dotnet/roslyn/issues/42814")]
+        public CSharpUsePatternCombinatorsCodeFixProvider()
+            : base(supportsFixAll: true)
+        {
+        }
+
+        private static SyntaxKind MapToSyntaxKind(BinaryOperatorKind kind)
+            => kind switch
+            {
+                BinaryOperatorKind.LessThan => SyntaxKind.LessThanToken,
+                BinaryOperatorKind.GreaterThan => SyntaxKind.GreaterThanToken,
+                BinaryOperatorKind.LessThanOrEqual => SyntaxKind.LessThanEqualsToken,
+                BinaryOperatorKind.GreaterThanOrEqual => SyntaxKind.GreaterThanEqualsToken,
+                _ => throw ExceptionUtilities.UnexpectedValue(kind)
+            };
+
+        public override ImmutableArray<string> FixableDiagnosticIds
+            => ImmutableArray.Create(IDEDiagnosticIds.UsePatternCombinatorsDiagnosticId);
+
+        internal sealed override CodeFixCategory CodeFixCategory => CodeFixCategory.CodeStyle;
+
+        public override Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            context.RegisterCodeFix(
+                new MyCodeAction(c => FixAsync(context.Document, context.Diagnostics.First(), c)),
+                context.Diagnostics);
+            return Task.CompletedTask;
+        }
+
+        protected override async Task FixAllAsync(
+            Document document, ImmutableArray<Diagnostic> diagnostics,
+            SyntaxEditor editor, CancellationToken cancellationToken)
+        {
+            var semanticModel = await document.RequireSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+            foreach (var diagnostic in diagnostics)
+            {
+                var location = diagnostic.Location;
+                var expression = editor.OriginalRoot.FindNode(location.SourceSpan, getInnermostNodeForTie: true);
+                var operation = semanticModel.GetOperation(expression, cancellationToken);
+                RoslynDebug.AssertNotNull(operation);
+                var pattern = CSharpUsePatternCombinatorsAnalyzer.Analyze(operation);
+                RoslynDebug.AssertNotNull(pattern);
+                var patternSyntax = AsPatternSyntax(pattern).WithAdditionalAnnotations(Formatter.Annotation);
```

Comment:
the annotation seems weird (since you are synthesizing patterns).  was it actually needed?

Reply:
since we're capturing trailing trivia, there will be some redundant trivia on the last node. 

I'll reconfirm if that's still the case if I missed something the first time.

Reply:
the original constants has some trivia which we don't want to remove and sometime it's redundant whitesapce.